sematic/plugins/metrics_storage/sql/sql_metrics_storage.py

```diff
@@ -168,23 +170,31 @@ def get_aggregated_metrics(
             select_fields.append(field_)
             group_by_clauses.append(field_)
 
-        with db().get_session() as session:
-            query = session.query(*select_fields).filter(*predicates)
+        try:
+            with db().get_session() as session:
+                query = session.query(*select_fields).filter(*predicates)
 
-            if len(group_by_clauses) > 0:
-                query = query.group_by(*group_by_clauses)
+                if len(group_by_clauses) > 0:
+                    query = query.group_by(*group_by_clauses)
 
-            if rollup == "auto":
-                field_ = func.extract("epoch", MetricValue.metric_time)
-                record_count = query.add_columns(field_).group_by(field_).count()
+                if rollup == "auto":
+                    field_ = func.extract("epoch", MetricValue.metric_time)
+                    record_count = query.add_columns(field_).group_by(field_).count()
 
-                if record_count > _MAX_SERIES_POINTS:
-                    field_ = field_ / interval_seconds * interval_seconds
+                    if record_count > _MAX_SERIES_POINTS:
+                        field_ = field_ / interval_seconds * interval_seconds
 
-                query = query.add_columns(field_).group_by(field_)
-                extra_field_names.append("timestamp")
+                    query = query.add_columns(field_).group_by(field_)
+                    extra_field_names.append("timestamp")
 
-            records = query.all()
+                records = query.all()
+        except sqlalchemy.exc.OperationalError as e:
+            # User has old SQLite version that does not support querying JSONB fields.
+            if str(e).startswith('(sqlite3.OperationalError) near ">>"'):
+                logger.error(SQLITE_WARNING_MESSAGE)
```

Comment:
This warning message says that Sematic will soon require the new version of sqlite, but it then fails the call by raising the error. This effectively makes the sqlite version required in order for the metrics feature to work locally.

Please use a different message that clarifies that this feature does not work locally without 3.38.0 instead.

Reply:
It does NOT fail the call, it returns empty metrics.